src/basicfuns.jl

```diff
@@ -308,10 +308,25 @@ $(SIGNATURES)
 Return `log(exp(x) + exp(y))`, avoiding intermediate overflow/undeflow, and handling
 non-finite values.
 """
-function logaddexp(x::Real, y::Real)
-    # ensure Δ = 0 if x = y = ± Inf
-    Δ = x == y ? zero(x - y) : abs(x - y)
-    max(x, y) + log1pexp(-Δ)
+logaddexp(x::Real, y::Real) = logaddexp(float.(promote(x, y))...)
+
+function logaddexp(x::T, y::T) where T <: AbstractFloat
```

Comment:
This is too strict, not every floating point number and output of `float` is of type `AbstractFloat`. Eg, usually dual numbers don't satisfy this property (typically they are subtypes of `Real` but not `AbstractFloat`).
Ie., this should be an internal method with signature

```suggestion
function _logaddexp(x::T, y::T) where {T<:Real}
```